src/Domain/Insights/FixerDecorator.php

```diff
@@ -136,6 +146,42 @@ public function addDiff(string $file, string $diff): void
         $this->errors[] = Details::make()->setFile($file)->setDiff($diff)->setMessage($diff);
     }
 
+    public function addFileFixed(SplFileInfo $fileInfo): void
+    {
+        $file = $fileInfo->getRelativePathname();
+        if (! \array_key_exists($file, $this->fixPerFile)) {
+            $this->fixPerFile[$file] = 0;
+        }
+
+        $this->fixPerFile[$file] = ++$this->fixPerFile[$file];
+        $this->totalFixed++;
+    }
+
+    public function getTotalFix(): int
+    {
+        return $this->totalFixed;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getFixPerFile(): array
+    {
+        $details = [];
+        foreach ($this->fixPerFile as $file => $count) {
+            $message = 'issues fixed';
+            if ($count === 1) {
+                $message = 'issue fixed';
+            }
+
+            $details[] = (new Details())
+                ->setMessage(sprintf('%s %s', $count, $message))
+                ->setFile($file);
+        }
+
+        return $details;
+    }
```

Comment:
Consider moving this and their variables inside a trait, as it is duplicated on `SniffDecorator`

Reply:
Done, I created a trait FixPerFileCollector